### PR TITLE
feat(21687): Add integration with Heap Analytics and user consent banner

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/21687/details/

The PR adds integration with the `Heap Analytics user journey tracking framework and a simple "Privacy Settings" consent form. 

The integration relies on an `application id` defined for the app (a different `id` for `production` and `development` environments), to be tracked by `Heap`. The implementation expects this `id` to be provided through the `VITE_MONITORING_HEAP` environment variables.  

Different strategies for open source or commercial installations must be investigated to provide the relevant `application id` to the app. 

### Design
- The design of the "Privacy Settings" banner is strongly inspired by the HiveMQ website UI; information related to advertisements has been removed from the description (Edge doesn't have any!)
- The "more information" link on the "Privacy Settings" banner links to https://www.hivemq.com/legal/imprint/
- Consent is designed to be **opt-in first**:
  - if the `application id` is not defined, tracking is not activated
  - if the consent is not actively denied or granted (e.g. closing the privacy banner), consent is not recorded and tracking is not activated
  - if the consent is denied, consent is recorded but tracking is not activated
  - if the consent is granted, consent is recorded and tracking is activated 
- Consent is structured to support multiple frameworks and individual opt-in. The UI to see and edit these services is not yet implemented
- Consent is stored in a `localStorage` key containing  every supported tracked frameworks 
 
### After
screenshot-localhost_3000-2024 04 30-17_51_31

### CODE REVIEW

File: hivemq-edge/src/frontend/index.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <link rel="icon" type="image/svg+xml" href="/hivemq-edge.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>HiveMQ Edge</title>
+    <script type="module" src="/src/trackers.ts"></script>
```
Comment (oli-hivemq) [inline hivemq-edge/src/frontend/index.html:8]: Have you checked that once the app is built and bundled, this location is still valid? I wouldn't have expected to see `/src` in there 🤔

File: hivemq-edge/src/frontend/src/components/PrivacyConsentBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+const PrivacyConsentBanner: FC = () => {
+  const { t } = useTranslation('components')
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const cancelRef = useRef<HTMLButtonElement>(null)
+  const [privacy, setPrivacy] = useLocalStorage<PrivacySourceGranted | undefined>('edge.privacy', undefined)
+
+  useEffect(() => {
+    // const headAppId = import.meta.env.VITE_MONITORING_HEAP
```
Comment (oli-hivemq) [inline hivemq-edge/src/frontend/src/components/PrivacyConsentBanner.tsx:32]: clean up?

File: hivemq-edge/src/frontend/index.html (hunk 1 above)
Comment (vanch3d) [reply hivemq-edge/src/frontend/index.html:8]: good point. It's not working

Comment (vanch3d) [reply hivemq-edge/src/frontend/index.html]: In fact, it is working (forgot to turn the adblocker off on the prod domain!). But I'm refactoring the approach for a better readability

**Comment by vanch3d** [comment]
Heap integration refactored to accommodate dual consent:
- Corporate overall consent (as specified by config/API property)
- End-user informed consent (as recorded in `localStorage`)
Note that the flow around the `<head>`-based script is just a glorified hack

File: hivemq-edge/src/frontend/src/locales/en/components.json [hunk 3]
```diff
@@ -61,6 +61,15 @@
     "action_COPIED": "Copied!",
     "action_ERROR": "Error!"
   },
+  "PrivacyConsentBanner": {
+    "header": "Privacy Settings",
+    "body": "This web application uses third-party website tracking technologies to provide and continually improve our services. I agree and may revoke or change my consent at any time with effect for the future.",
```
Comment (oli-hivemq) [inline hivemq-edge/src/frontend/src/locales/en/components.json:66]: I find the last sentence a little confusing.
Proposal for an alternative: "This web application uses third-party tracking technologies to continually improve our services. You can agree to their use now and change or revoke your consent at any time in the future."

File: hivemq-edge/src/frontend/.env [hunk 4]
```diff
@@ -6,3 +6,6 @@ VITE_FLAG_MOCK_SERVER=false
 VITE_FLAG_FACET_SEARCH=true
 VITE_FLAG_WORKSPACE_FLOW_PANEL=true
 VITE_FLAG_TOPIC_EDITOR_SHOW_BRANCHES=false
+
+### User Analytics
+VITE_MONITORING_HEAP=3411251519
```
Comment (oli-hivemq) [inline hivemq-edge/src/frontend/.env:11]: Is it a value that we're comfortable with being present in the repo?

File: hivemq-edge/src/frontend/.prettierignore [hunk 5]
```diff
@@ -6,4 +6,5 @@ dist/
 coverage/
 public/
 src/api/__generated__
+src/trackers.ts
```
Comment (oli-hivemq) [inline hivemq-edge/src/frontend/.prettierignore:9]: This file was removed, wasn't it?

File: hivemq-edge/src/frontend/.env (hunk 4 above)
Comment (vanch3d) [reply hivemq-edge/src/frontend/.env:11]: Yes, its the default for the production tracker

**Review by sfrehse — APPROVED** [review]

**Pull request merged**